### ISSUE TITLE
How do I convert Windows path to Linux?

### ISSUE THREAD

**Issue by SamTyurenkov** [mtbdata711/vscode-phpsab-docker#12]
Hi there, I set up your extension on macOS first and it was working.
Now I want to set up it on Windows and there are some paths issues:

`ERROR: The file "/var/www/site/\web\app\themes\theme\functions.php" does not exist. Run "phpcs --help" for usage information SyntaxError: Unexpected token 'E', "ERROR: The"... is not valid JSON`

site is a mounted volume to php-fpm-custom container at /var/www/site

My config is 

```
    "phpsab.docker.dockerExecutablePathCS": "/var/www/site/vendor/bin/phpcs",
    "phpsab.docker.dockerExecutablePathCBF": "/var/www/site/vendor/bin/phpcbf",
    "phpsab.docker.snifferMode": "onType",
    "phpsab.docker.dockerContainer": "php-fpm-custom",
    "phpsab.docker.dockerWorkspaceRoot": "/var/www/site/",
    "phpsab.docker.dockerEnabled": true,
    "php.validate.enable": false,
    "php.validate.executablePath": "",
    "phpsab.docker.allowedAutoRulesets": [
        ".phpcs.xml",
        ".phpcs.xml.dist",
        "phpcs.xml",
        "phpcs.xml.dist",
        "phpcs.ruleset.xml",
        "ruleset.xml"
    ],
    "phpsab.docker.executablePathCS": "",
    "phpsab.docker.executablePathCBF": "",
    "phpsab.docker.snifferShowSources": true,
    "phpsab.docker.containerExec": "docker",
     "phpsab.docker.standard": ""
```

I use git bash as default vscode terminal

I also tried 

`    "phpsab.docker.containerExec": "MSYS_NO_PATHCONV=1 docker"`

As I need it to run docker from git bash with $(pwd) commands. But it didnt help.

**Comment by SamTyurenkov** [comment]
another error that might appear is:

`ERROR: the "/var/www/site/\.phpcs.xml" coding standard is not installed. The installed coding standards are MySource, PEAR, PSR1, PSR2, PSR12, Squiz, Zend, Modernize, NormalizedArrays, Universal, PHPCSUtils, WordPress, WordPress-Core, WordPress-Docs and WordPress-Extra SyntaxError: Unexpected token 'E', "ERROR: the"... is not valid JSON`

**Comment by SamTyurenkov** [comment]
and also, if I manually specify standard, it still processes the path incorrectly

```
An error occurred during processing; checking has been aborted. The error message was: fileperms(): stat failed for /var/www/site/\web\app\themes\theme\functions.php in /var/www/site/vendor/squizlabs/php_codesniffer/src/Standards/Generic/Sniffs/Files/ExecutableFileSniff.php on line 48\nThe error originated in the Generic.Files.ExecutableFile sniff on line 48.
(Internal.Exception)
```

**Comment by mtbdata711** [comment]
Hey @SamTyurenkov,

I'll look into this when I get some downtime from work.

I'm not a Windows user myself so I've not come across the bug before. Let me know if you get a chance to pull the code down and see if you can spot the fix.

Cheers

**Comment by SamTyurenkov** [comment]
> Hey @SamTyurenkov,
> 
> I'll look into this when I get some downtime from work.
> 
> I'm not a Windows user myself so I've not come across the bug before. Let me know if you get a chance to pull the code down and see if you can spot the fix.
> 
> Cheers

I can try, but would you please explain how do I make sure that in extension development host i run a just built version and not production one? in extensions list it looks like the one installed is used, and not the one I launched. Is there a way to distinct it it visually somewhere? 

Also where should I expect to see the debug info, e.g. if I add console.log somewhere in a consrtructor, where should I expect to see it in vscode ui, output->EXTENSION_NAME or output->extension host or somewhere else?

**Comment by mtbdata711** [comment]
Hey @SamTyurenkov,

You should be able to enter the extension debug mode by pressing F5. 

Once you have done this and you have opened a PHP project to test against you should see any console.logs in the Debug console in the VS Code window of vscode-phpsab-docker. 

I'd suggest that you disable the installed extension while you are doing these tests.

I've created a branch which should resolve this here: https://github.com/mtbdata711/vscode-phpsab-docker/tree/mtbdata711/fix/support-windows-paths-issue-12

Let me know if you are able to pull it down and give it a test drive! 

Cheers

**Comment by SamTyurenkov** [comment]
I will give it a try. Thank you.
Already looked briefly into the debug with my own attempt, but didnt see my console.log string, so I thought I did something wrong.

Do you have to reload vscode window after every build?

**Comment by mtbdata711** [comment]
Hey @SamTyurenkov,

The debugging setup could be improved at some point! 

You can restart the extension window by pressing Ctrl+Shift+F5 or the restart icon in the debug bar in the top right of VS Code. This should also rebuild all of your assets.

If you add `{"phpsab.docker.debug": true}` to your settings.json you will be able to see logs dispatched by `logger.ts` in the Output console in the extension window which is very useful for seeing the calls made to phpcs / phpcbf and your configuration (see screenshot). 

![Screenshot from 2024-01-26 22-52-31](https://github.com/mtbdata711/vscode-phpsab-docker/assets/44591353/e7e86d0e-ef59-4dab-a2f7-7377286d6e8a)

Let me know if you are able to test. Otherwise, I'm happy to publish the update as it is not a breaking change if that is easier. 

Cheers

**Comment by SamTyurenkov** [comment]
@mtbdata711 

for some reason launching in extension development host doesn't pick up the workspace settings with paths. It keeps looking for phpcs and phpcbf on host machine, while I set "use docker" to enabled and added docker paths to phpcs and phpcbf. I tried also with user settings.

also I had to update webpack, as it was failing to launch with some library error.

![image](https://github.com/mtbdata711/vscode-phpsab-docker/assets/19373777/7cc250a9-cf31-42b4-b052-37c4a5d30eb1)

**Comment by mtbdata711** [comment]
Hey @SamTyurenkov,

What Node version are you running? I am using v16.14.2 and not having an issue with Webpack. 

Will make a note to add an .nvmrc file at some point.

I think this may be a bug in how the executable paths are set, could you run again with the following settings:

`{
    "phpsab.docker.dockerContainer": "php-fpm-custom",
    "phpsab.docker.snifferMode": "onType",
    "phpsab.docker.dockerEnabled": true,
    "phpsab.docker.debug": true,
    "phpsab.docker.executablePathCS": "<ABSOLUTE_PATH_TO_PHPCS_ON_LOCAL_SYSTEM>",
    "phpsab.docker.executablePathCBF": "<ABSOLUTE_PATH_TO_PHPCBF_ON_LOCAL_SYSTEM>"
}`

and copy your configuration which should be printed in the Output console of the Extension Development Host VS Code Window. 

This should give us enough to go on to begin to put in a fix. 

Setting `phpsab.docker.executablePathCBF` and `phpsab.docker.executablePathCBF` to the paths on your local system should get around the validation steps which are looking for the .bat Windows executable where none exists.

Cheers

**Comment by mtbdata711** [comment]
Hey @SamTyurenkov,

What Node version are you running? I am using v16.14.2 and not having an issue with Webpack. 

Will make a note to add an .nvmrc file at some point.

I think this may be a bug in how the executable paths are set, could you run again with the following settings:

`{
    "phpsab.docker.dockerContainer": "php-fpm-custom",
    "phpsab.docker.snifferMode": "onType",
    "phpsab.docker.dockerEnabled": true,
    "phpsab.docker.debug": true
}`

and copy your configuration which should be printed in the Output console of the Extension Development Host VS Code Window. This should give us enough to go on to begin to put in a fix. 

Cheers

**Comment by SamTyurenkov** [comment]
@mtbdata711 I have node 18+ and 20+.

Your suggestion almost worked. Well actually it works without specifying ruleset manually or automatically.
 
Auto Ruleset Search or Allowed Auto Rulesets if set result in this error:
`ERROR: the "e:/WWW/localhost/phpcs.xml" coding standard is not installed. The installed coding standards are MySource, PEAR, PSR1, PSR2, PSR12, Squiz, Zend, Modernize, NormalizedArrays, Universal, PHPCSUtils, WordPress, WordPress-Core, WordPress-Docs and WordPress-Extra SyntaxError: Unexpected token 'E', "ERROR: the"... is not valid JSON`

If I set Standard manually the error is almost same:
`ERROR: the "./phpcs.xml" coding standard is not installed. The installed coding standards are MySource, PEAR, PSR1, PSR2, PSR12, Squiz, Zend, Modernize, NormalizedArrays, Universal, PHPCSUtils, WordPress, WordPress-Core, WordPress-Docs and WordPress-Extra SyntaxError: Unexpected token 'E', "ERROR: the"... is not valid JSON`

in both cases it looks like its picking it on host machine, not docker. Is there a way to set auto ruleset search or manual ruleset with a docker path?

**Comment by mtbdata711** [comment]
Hey @SamTyurenkov,

Glad to hear we are making progress on this! Is everything working if you specify one of your installed standards, such as "WordPress-Extra"? 

What configuration do you have in your settings.json file?

By default, with no `phpsab.docker.standard` option set and `phpsab.docker.autoRulesetSearch` set to true then your standard should be detected as a Windows path and then formatted to Docker.

This is done in `src/resolvers/standards-path-resolver.ts`.  The found paths are sent to `DockerPathResolver.resolveDocker` and converted to match your Docker path, i.e. if your path is  `e:/WWW/localhost/.phpcs.xml` then it should become `/var/www/localhost/.phpcs.xml`.

Alternatively, you should be able update your configuration to:

`{ "phpsab.docker.standard": "/var/www/localhost/.phpcs.xml",
    "phpsab.docker.autoRulesetSearch": false }`

and it should work. 

Under the hood, this will generate and run the command below: 

`docker exec -i ${phpsab.docker.dockerContainer} ${phpsab.docker.dockerExecutablePathCS} --report=json -q --standard=${phpsab.docker.standard} --stdin-path=${DockerPathToCurrentFile} -`

For my Test instance this looks like the following:

`docker exec -i 39fd3a7a3d4d /var/www/html/vendor/bin/phpcs --report=json -q --standard=/var/www/html/.phpcs.xml --stdin-path=/var/www/html/index.php -`

Are you on the development branch I sent over - https://github.com/mtbdata711/vscode-phpsab-docker/tree/mtbdata711/fix/support-windows-paths-issue-12 - and able to provide logs from the Output console?

Cheers

**Comment by SamTyurenkov** [comment]
@mtbdata711 

yeah, I think i tried all ways. 
auto search (error above), 
auto rules (error above), 
manual relative (error above), 
manual docker path (no error, but it just doesnt use it and uses default rules, not from my phpcs.xml)

so why is there JSON error anyway, I posted above? phpcs.xml is not a JSON obviously, it looks like it is added somewhere incorrectly, in some JSON and the way it is added is not a JSON format right?

Yeah I can post output, but there is nothing useful, as for settings.json in it (I did change/remove some of them as I mentioned, but here are current ones)

```
["INFO" - 11:29:23 PM] CONFIGURATION
{
  "resources": [
    {
      "useFilepath": false,
      "containerExec": "docker",
      "fixerEnable": true,
      "fixerArguments": [],
      "workspaceRoot": "e:\\WWW\\localhost",
      "executablePathCBF": "e:\\WWW\\localhost",
      "executablePathCS": "e:\\WWW\\localhost",
      "composerJsonPath": "composer.json",
      "standard": "/var/www/localhost/phpcs.xml",
      "autoRulesetSearch": true,
      "allowedAutoRulesets": [
        ".phpcs.xml",
        ".phpcs.xml.dist",
        "phpcs.xml",
        "phpcs.xml.dist",
        "phpcs.ruleset.xml",
        "ruleset.xml"
      ],
      "snifferEnable": true,
      "snifferArguments": [],
      "dockerEnabled": true,
      "dockerContainer": "php-fpm-custom",
      "dockerWorkspaceRoot": "/var/www/localhost",
      "dockerExecutablePathCBF": "/var/www/localhost/vendor/bin/phpcbf",
      "dockerExecutablePathCS": "/var/www/localhost/vendor/bin/phpcs"
    }
  ],
  "snifferMode": "onType",
  "snifferShowSources": false,
  "snifferTypeDelay": 250,
  "debug": true
}
["INFO" - 11:29:23 PM] Sniffer running
["INFO" - 11:29:23 PM] Standards Search paths: 
[
  "e:\\WWW\\localhost\\"
]
["INFO" - 11:29:23 PM] DOCKER: Resolved docker path: e:/WWW/localhost/phpcs.xml
["INFO" - 11:29:23 PM] DOCKER: Resolved docker path: e:/WWW/localhost/index.php
["INFO" - 11:29:23 PM] SNIFFER COMMAND: docker exec -i php-fpm-custom /var/www/localhost/vendor/bin/phpcs --report=json -q --standard=e:/WWW/localhost/phpcs.xml --stdin-path=e:/WWW/localhost/index.php -
["INFO" - 11:29:23 PM] Sniffer ran for 0.006 seconds
["ERROR" - 11:29:23 PM] ERROR: the "e:/WWW/localhost/phpcs.xml" coding standard is not installed. The installed coding standards are MySource, PEAR, PSR1, PSR2, PSR12, Squiz, Zend, Modernize, NormalizedArrays, Universal, PHPCSUtils, WordPress, WordPress-Core, WordPress-Docs and WordPress-Extra

SyntaxError: Unexpected token 'E', "ERROR: the"... is not valid JSON
```

**Comment by SamTyurenkov** [comment]
Here is the start without automatic resolve. There is no error, but it just silently not uses the path to phpcs.xml (if it is a docker path). if it is a host path, it gives above error.

```
["INFO" - 12:50:35 AM] CONFIGURATION
{
  "resources": [
    {
      "useFilepath": false,
      "containerExec": "docker",
      "fixerEnable": true,
      "fixerArguments": [],
      "workspaceRoot": "e:\\WWW\\localhost",
      "executablePathCBF": "e:\\WWW\\localhost",
      "executablePathCS": "e:\\WWW\\localhost",
      "composerJsonPath": "composer.json",
      "standard": "/var/www/localhost/phpcs.xml",
      "autoRulesetSearch": false,
      "allowedAutoRulesets": [
        ".phpcs.xml",
        ".phpcs.xml.dist",
        "phpcs.xml",
        "phpcs.xml.dist",
        "phpcs.ruleset.xml",
        "ruleset.xml"
      ],
      "snifferEnable": true,
      "snifferArguments": [],
      "dockerEnabled": true,
      "dockerContainer": "php-fpm-custom",
      "dockerWorkspaceRoot": "/var/www/localhost",
      "dockerExecutablePathCBF": "/var/www/localhost/vendor/bin/phpcbf",
      "dockerExecutablePathCS": "/var/www/localhost/vendor/bin/phpcs"
    }
  ],
  "snifferMode": "onType",
  "snifferShowSources": false,
  "snifferTypeDelay": 250,
  "debug": true
}
["INFO" - 12:50:35 AM] Sniffer running
["INFO" - 12:50:35 AM] DOCKER: Resolved docker path: e:/WWW/localhost/index.php
["INFO" - 12:50:35 AM] SNIFFER COMMAND: docker exec -i php-fpm-custom /var/www/localhost/vendor/bin/phpcs --report=json -q --standard=/var/www/localhost/phpcs.xml --stdin-path=e:/WWW/localhost/index.php -
["INFO" - 12:50:35 AM] Sniffer ran for 0.007 seconds
["INFO" - 12:50:40 AM] DOCKER: Resolved local path: e:\WWW\localhost\index.php
```

**Comment by SamTyurenkov** [comment]
See how IDE only highlights default phpcs rules on upper side of the screen. 
And note that in terminal below, there is a correct sniff with a lot more errors if phpcs.xml is used 

I have alias for phpcs command which is = `alias phpcs="MSYS_NO_PATHCONV=1 docker run --rm -it -v $(pwd):/app -w /app php-fpm-custom vendor/bin/phpcs"`

![image](https://github.com/mtbdata711/vscode-phpsab-docker/assets/19373777/f92adeb4-c419-4911-b20e-8a97d283162a)

**Comment by mtbdata711** [comment]
Hey @SamTyurenkov,

Cheers for doing a deep dive into this. As I said, I'm not a Windows user so getting the logs from your side is a big help. 

I've updated the branch, can you pull in the updates and try again and see if that resolves it?

It looks like the `document.fileName` value coming in from the `TextDocument` type from the `vscode` module normalises all paths into unix style paths so the `resolveDocker` and `resolveLocal` methods had to be reverted to account for this.

You can see this in the generated command not resolving the filename correctly and instead using `e:/WWW/localhost/index.php`:

`docker exec -i php-fpm-custom /var/www/localhost/vendor/bin/phpcs --report=json -q --standard=e:/WWW/localhost/phpcs.xml --stdin-path=e:/WWW/localhost/index.php -`

The JSON you are seeing is the error output from phpcs as it cannot find the standards file (notice that it can when it fails silently as the filename is still wrong). 

If you pull down the updates and run with the following configuration then I'd expect everything to work. 

I would be interested in getting another dump of the logs after you have done this, alongside one testing `"phpsab.docker.autoRulesetSearch":true`.

`
{
	"phpsab.docker.dockerEnabled": true,
	"phpsab.docker.dockerContainer": "php-fpm-custom",
	"phpsab.docker.dockerWorkspaceRoot": "/var/www/localhost",
	"phpsab.docker.executablePathCS": "E:\\WWW\\localhost\\vendor\\bin\\phpcs",
	"phpsab.docker.executablePathCBF": "E:\\WWW\\localhost\\vendor\\bin\\phpcbf",
	"phpsab.docker.standard": "/var/www/localhost/phpcs.xml",
	"phpsab.docker.debug": true,
	"phpsab.docker.autoRulesetSearch": false,
}  
`

Cheers!

**Comment by mtbdata711** [comment]
Hey @SamTyurenkov,

~~Just done a further patch so you may need to pull again.~~  

Ignore this, few more tweaks to iron out. Sorry about that!


Cheers!

**Comment by SamTyurenkov** [comment]
#Just done a further patch so you may need to pull again.

meanwhile, auto resolver still has path error in the last part: 
`ERROR: the "/var/www/localhost\phpcs.xml"....`

as for your specific config, did you want an escape in paths? Windows paths work normally in VScode settings usually. 
If I need a windows path, I just use relative or /e/WWW/, some other extensions correctly worked with /e/WWW path for example. 
However in console, it represents those paths Windows style E:\WWW\

Actually, Im not sure about other extension, it might be on other PC. But I have my repository with docker compose, where I have .env this way:

```
MYSQL_ROOT_PASSWORD=root
WWW_PATH=/e/WWW
SSL_PATH=/e/SSL
DB_PATH=/e/MARIADB
NGINX_CONF_PATH=/e/NGINXCONF
```

and in install.sh it is used this way:

```
#!/bin/bash

# Source the .env file to load environment variables
if [ -f .env ]; then
    export $(cat .env | xargs)
fi

SSL_PATH=$(eval echo $SSL_PATH)
WWW_PATH=$(eval echo $WWW_PATH)
DB_PATH=$(eval echo $DB_PATH)
NGINX_CONF_PATH=$(eval echo $NGINX_CONF_PATH)
```

These paths are than used for Volume creation and mounting in containers for persistent data.

**Comment by SamTyurenkov** [comment]
On a second thought, this might be actually dependent on terminal which is currently used. I run it with bash, but vscode might pick powershell for internal jobs.

**Comment by mtbdata711** [comment]
Hey @SamTyurenkov,

Just pushed up a fix that should be a bit more flexible and either format should work, in theory at least.

Can you give it a go and post any logs here if the standards issue regarding auto resolving is still occuring. I'm not fully confident that is fixed.

Cheers for your help on this!

**Comment by mtbdata711** [comment]
@SamTyurenkov ,

Another patch gone in, hand't factored `document.uri.fsPath` doesn't do any case changes apart from the drive name and this will mess with the value of `normalisedWorkspaceRoot`.

Cheers

**Comment by SamTyurenkov** [comment]
@mtbdata711 seems to be working now.

![image](https://github.com/mtbdata711/vscode-phpsab-docker/assets/19373777/40bd3a06-0e0d-44fc-b973-2e53d04c3057)


```
["INFO" - 9:33:14 AM] CONFIGURATION
{
  "resources": [
    {
      "useFilepath": false,
      "containerExec": "docker",
      "fixerEnable": true,
      "fixerArguments": [],
      "workspaceRoot": "e:\\WWW\\localhost",
      "executablePathCBF": "E:\\WWW\\localhost\\vendor\\bin\\phpcbf",
      "executablePathCS": "E:\\WWW\\localhost\\vendor\\bin\\phpcs",
      "composerJsonPath": "composer.json",
      "standard": null,
      "autoRulesetSearch": true,
      "allowedAutoRulesets": [
        ".phpcs.xml",
        ".phpcs.xml.dist",
        "phpcs.xml",
        "phpcs.xml.dist",
        "phpcs.ruleset.xml",
        "ruleset.xml"
      ],
      "snifferEnable": true,
      "snifferArguments": [],
      "dockerEnabled": true,
      "dockerContainer": "php-fpm-custom",
      "dockerWorkspaceRoot": "/var/www/localhost",
      "dockerExecutablePathCBF": "/var/www/localhost/vendor/bin/phpcbf",
      "dockerExecutablePathCS": "/var/www/localhost/vendor/bin/phpcs"
    }
  ],
  "snifferMode": "onType",
  "snifferShowSources": false,
  "snifferTypeDelay": 250,
  "debug": true
}
["INFO" - 9:33:14 AM] Sniffer running
["INFO" - 9:33:14 AM] Standards Search paths: 
[
  "e:\\WWW\\localhost\\"
]
["INFO" - 9:33:14 AM] Standards Search files: 
[
  "e:\\WWW\\localhost\\.phpcs.xml",
  "e:\\WWW\\localhost\\.phpcs.xml.dist",
  "e:\\WWW\\localhost\\phpcs.xml",
  "e:\\WWW\\localhost\\phpcs.xml.dist",
  "e:\\WWW\\localhost\\phpcs.ruleset.xml",
  "e:\\WWW\\localhost\\ruleset.xml"
]
["INFO" - 9:33:14 AM] Standards Search found: 
"e:\\WWW\\localhost\\phpcs.xml"
["INFO" - 9:33:14 AM] DOCKER: Resolved docker path: /var/www/localhost/phpcs.xml
["INFO" - 9:33:14 AM] DOCKER: Resolved docker path: /var/www/localhost/index.php
["INFO" - 9:33:14 AM] SNIFFER COMMAND: docker exec -i php-fpm-custom /var/www/localhost/vendor/bin/phpcs --report=json -q --standard=/var/www/localhost/phpcs.xml --stdin-path=/var/www/localhost/index.php -
["INFO" - 9:33:14 AM] Sniffer ran for 0.01 seconds
["INFO" - 9:33:19 AM] DOCKER: Resolved local path: e:\WWW\localhost/index.php
["INFO" - 9:33:34 AM] CONFIGURATION
{
  "resources": [
    {
      "useFilepath": false,
      "containerExec": "docker",
      "fixerEnable": true,
      "fixerArguments": [],
      "workspaceRoot": "e:\\WWW\\localhost",
      "executablePathCBF": "E:\\WWW\\localhost\\vendor\\bin\\phpcbf",
      "executablePathCS": "E:\\WWW\\localhost\\vendor\\bin\\phpcs",
      "composerJsonPath": "composer.json",
      "standard": null,
      "autoRulesetSearch": true,
      "allowedAutoRulesets": [
        ".phpcs.xml",
        ".phpcs.xml.dist",
        "phpcs.xml",
        "phpcs.xml.dist",
        "phpcs.ruleset.xml",
        "ruleset.xml"
      ],
      "snifferEnable": true,
      "snifferArguments": [],
      "dockerEnabled": true,
      "dockerContainer": "php-fpm-custom",
      "dockerWorkspaceRoot": "/var/www/localhost",
      "dockerExecutablePathCBF": "/var/www/localhost/vendor/bin/phpcbf",
      "dockerExecutablePathCS": "/var/www/localhost/vendor/bin/phpcs"
    }
  ],
  "snifferMode": "onType",
  "snifferShowSources": false,
  "snifferTypeDelay": 250,
  "debug": true
}
["INFO" - 9:33:34 AM] CONFIGURATION
{
  "resources": [
    {
      "useFilepath": false,
      "containerExec": "docker",
      "fixerEnable": true,
      "fixerArguments": [],
      "workspaceRoot": "e:\\WWW\\localhost",
      "executablePathCBF": "E:\\WWW\\localhost\\vendor\\bin\\phpcbf",
      "executablePathCS": "E:\\WWW\\localhost\\vendor\\bin\\phpcs",
      "composerJsonPath": "composer.json",
      "standard": null,
      "autoRulesetSearch": true,
      "allowedAutoRulesets": [
        ".phpcs.xml",
        ".phpcs.xml.dist",
        "phpcs.xml",
        "phpcs.xml.dist",
        "phpcs.ruleset.xml",
        "ruleset.xml"
      ],
      "snifferEnable": true,
      "snifferArguments": [],
      "dockerEnabled": true,
      "dockerContainer": "php-fpm-custom",
      "dockerWorkspaceRoot": "/var/www/localhost",
      "dockerExecutablePathCBF": "/var/www/localhost/vendor/bin/phpcbf",
      "dockerExecutablePathCS": "/var/www/localhost/vendor/bin/phpcs"
    }
  ],
  "snifferMode": "onType",
  "snifferShowSources": false,
  "snifferTypeDelay": 250,
  "debug": true
}
["INFO" - 9:33:34 AM] Sniffer running
["INFO" - 9:33:34 AM] Standards Search paths: 
[
  "e:\\WWW\\localhost\\"
]
["INFO" - 9:33:34 AM] Standards Search files: 
[
  "e:\\WWW\\localhost\\.phpcs.xml",
  "e:\\WWW\\localhost\\.phpcs.xml.dist",
  "e:\\WWW\\localhost\\phpcs.xml",
  "e:\\WWW\\localhost\\phpcs.xml.dist",
  "e:\\WWW\\localhost\\phpcs.ruleset.xml",
  "e:\\WWW\\localhost\\ruleset.xml"
]
["INFO" - 9:33:34 AM] Standards Search found: 
"e:\\WWW\\localhost\\phpcs.xml"
["INFO" - 9:33:34 AM] DOCKER: Resolved docker path: /var/www/localhost/phpcs.xml
["INFO" - 9:33:34 AM] DOCKER: Resolved docker path: /var/www/localhost/index.php
["INFO" - 9:33:34 AM] SNIFFER COMMAND: docker exec -i php-fpm-custom /var/www/localhost/vendor/bin/phpcs --report=json -q --standard=/var/www/localhost/phpcs.xml --stdin-path=/var/www/localhost/index.php -
["INFO" - 9:33:34 AM] Sniffer ran for 0.005 seconds
["INFO" - 9:33:39 AM] DOCKER: Resolved local path: e:\WWW\localhost/index.php
["INFO" - 9:33:46 AM] CONFIGURATION
{
  "resources": [
    {
      "useFilepath": false,
      "containerExec": "docker",
      "fixerEnable": true,
      "fixerArguments": [],
      "workspaceRoot": "e:\\WWW\\localhost",
      "executablePathCBF": "E:\\WWW\\localhost\\vendor\\bin\\phpcbf",
      "executablePathCS": "E:\\WWW\\localhost\\vendor\\bin\\phpcs",
      "composerJsonPath": "composer.json",
      "standard": null,
      "autoRulesetSearch": true,
      "allowedAutoRulesets": [
        ".phpcs.xml",
        ".phpcs.xml.dist",
        "phpcs.xml",
        "phpcs.xml.dist",
        "phpcs.ruleset.xml",
        "ruleset.xml"
      ],
      "snifferEnable": true,
      "snifferArguments": [],
      "dockerEnabled": true,
      "dockerContainer": "php-fpm-custom",
      "dockerWorkspaceRoot": "/var/www/localhost",
      "dockerExecutablePathCBF": "/var/www/localhost/vendor/bin/phpcbf",
      "dockerExecutablePathCS": "/var/www/localhost/vendor/bin/phpcs"
    }
  ],
  "snifferMode": "onType",
  "snifferShowSources": true,
  "snifferTypeDelay": 250,
  "debug": true
}
["INFO" - 9:33:46 AM] CONFIGURATION
{
  "resources": [
    {
      "useFilepath": false,
      "containerExec": "docker",
      "fixerEnable": true,
      "fixerArguments": [],
      "workspaceRoot": "e:\\WWW\\localhost",
      "executablePathCBF": "E:\\WWW\\localhost\\vendor\\bin\\phpcbf",
      "executablePathCS": "E:\\WWW\\localhost\\vendor\\bin\\phpcs",
      "composerJsonPath": "composer.json",
      "standard": null,
      "autoRulesetSearch": true,
      "allowedAutoRulesets": [
        ".phpcs.xml",
        ".phpcs.xml.dist",
        "phpcs.xml",
        "phpcs.xml.dist",
        "phpcs.ruleset.xml",
        "ruleset.xml"
      ],
      "snifferEnable": true,
      "snifferArguments": [],
      "dockerEnabled": true,
      "dockerContainer": "php-fpm-custom",
      "dockerWorkspaceRoot": "/var/www/localhost",
      "dockerExecutablePathCBF": "/var/www/localhost/vendor/bin/phpcbf",
      "dockerExecutablePathCS": "/var/www/localhost/vendor/bin/phpcs"
    }
  ],
  "snifferMode": "onType",
  "snifferShowSources": true,
  "snifferTypeDelay": 250,
  "debug": true
}
["INFO" - 9:33:46 AM] Sniffer running
["INFO" - 9:33:46 AM] Standards Search paths: 
[
  "e:\\WWW\\localhost\\"
]
["INFO" - 9:33:46 AM] Standards Search files: 
[
  "e:\\WWW\\localhost\\.phpcs.xml",
  "e:\\WWW\\localhost\\.phpcs.xml.dist",
  "e:\\WWW\\localhost\\phpcs.xml",
  "e:\\WWW\\localhost\\phpcs.xml.dist",
  "e:\\WWW\\localhost\\phpcs.ruleset.xml",
  "e:\\WWW\\localhost\\ruleset.xml"
]
["INFO" - 9:33:46 AM] Standards Search found: 
"e:\\WWW\\localhost\\phpcs.xml"
["INFO" - 9:33:46 AM] DOCKER: Resolved docker path: /var/www/localhost/phpcs.xml
["INFO" - 9:33:46 AM] DOCKER: Resolved docker path: /var/www/localhost/index.php
["INFO" - 9:33:46 AM] SNIFFER COMMAND: docker exec -i php-fpm-custom /var/www/localhost/vendor/bin/phpcs --report=json -q --standard=/var/www/localhost/phpcs.xml --stdin-path=/var/www/localhost/index.php -
["INFO" - 9:33:46 AM] Sniffer ran for 0.005 seconds
["INFO" - 9:33:51 AM] DOCKER: Resolved local path: e:\WWW\localhost/index.php
["INFO" - 9:33:55 AM] Sniffer running
["INFO" - 9:33:55 AM] Standards Search paths: 
[
  "e:\\WWW\\localhost\\"
]
["INFO" - 9:33:55 AM] Standards Search files: 
[
  "e:\\WWW\\localhost\\.phpcs.xml",
  "e:\\WWW\\localhost\\.phpcs.xml.dist",
  "e:\\WWW\\localhost\\phpcs.xml",
  "e:\\WWW\\localhost\\phpcs.xml.dist",
  "e:\\WWW\\localhost\\phpcs.ruleset.xml",
  "e:\\WWW\\localhost\\ruleset.xml"
]
["INFO" - 9:33:55 AM] Standards Search found: 
"e:\\WWW\\localhost\\phpcs.xml"
["INFO" - 9:33:55 AM] DOCKER: Resolved docker path: /var/www/localhost/phpcs.xml
["INFO" - 9:33:55 AM] DOCKER: Resolved docker path: /var/www/localhost/index.php
["INFO" - 9:33:55 AM] SNIFFER COMMAND: docker exec -i php-fpm-custom /var/www/localhost/vendor/bin/phpcs --report=json -q --standard=/var/www/localhost/phpcs.xml --stdin-path=/var/www/localhost/index.php -
["INFO" - 9:33:55 AM] Sniffer ran for 0.005 seconds
["INFO" - 9:34:00 AM] DOCKER: Resolved local path: e:\WWW\localhost/index.php
```

**Comment by mtbdata711** [comment]
Hey @SamTyurenkov,

Sorry for the delay.

Great to hear it is working now! I've just published a new version (0.0.8) with the fix included.

Let me know how it goes!
